### PR TITLE
MOBILE-7013: Introduce ability to set custom host for pbs

### PR DESCRIPTION
**Pull request by KostiantynTselykh**
feat(PrebidRenderingSettings):
- Added ability to change bidServerHost

refactoring(DemoRenderingKotlin)
- Replaced reflection usage for setting bid server host

tests:
- Added unit test to verify that bid server host was changed (if not empty)

### CODE REVIEW

File: PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java [hunk 1]
```diff
@@ -75,6 +73,7 @@
 
     private static SdkInitListener sInitSdkListener;
 
+    private static String sBidServerHost = "https://prebid.openx.net/openrtb2/auction";
```
Comment (YuriyVelichkoOpenX) [inline PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java]: Need to remove this URL. OpenX is a custom server for Prebid. Can we mimic the host initialization behaviour from Prebid Core?

**Review by YuriyVelichkoOpenX — CHANGES_REQUESTED** [review]

File: PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java (hunk 1 above)
Comment (KostiantynTselykh) [reply PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java]: Will do

**Comment by KostiantynTselykh** [comment]
@YuriyVelichkoOpenX Applied changes to mimic prebid core host setup.
Also renamed demo app module to: PrebidInternalTestApp.

File: PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java [hunk 2]
```diff
@@ -91,7 +90,7 @@
      */
 
     public static void initializeSDK(Context context, final SdkInitListener sdkInitListener)
-    throws AdException {
+            throws AdException {
         Log.d(TAG, "Initializing OpenX SDK");
```
Comment (YuriyVelichkoOpenX) [inline PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java:94]: In the scope of future work need to review appearance of the words "OpenX" and "Apollo" one more time and remove them.

**Review by YuriyVelichkoOpenX — APPROVED** [review]
LGTM
+ short comment
+ make sure that all tests  work

File: PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java (hunk 2 above)
Comment (KonstantinTselykh) [reply PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java:94]: Already in progress in a separate branch.

Comment (KostiantynTselykh) [reply PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java:94]: Already in progress in a separate branch.

**Pull request merged**